test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -186,7 +187,7 @@ public void IrisVectorLightGbmWithLoadColumnName()
             }).PredictedLabel);
         }
 
-        [Fact]
+        [NativeDependencyFact("SQLite.Interop")]
```

Comment:
1. Why just this test? There are other tests in this class that depend on SQLite.
2. Does this cause the test to not run on Windows anymore? We don't use SQLite on windows - only on linux and mac. See:

```
        /// <summary>
        /// Non-Windows builds do not support SqlClientFactory/MSSQL databases. Hence, an equivalent
        /// SQLite database is used on Linux and MacOS builds.
        /// </summary>
        /// <returns>Return the appropiate Iris DatabaseSource according to build OS.</returns>
        private DatabaseSource GetIrisDatabaseSource(string command, int commandTimeoutInSeconds = 30)
```

Reply:
1 - The SQLite package is a mix of managed and native code. The managed code runs just fine everywhere, but the native code doesn't have a package for all RIDs, specifically arm/64.

2 - The `NativeDependencyFact` checks for the presence of the DLL and if its found it runs the test, otherwise it skips. Windows does have `SQLite.Interop`, so even though we aren't using that file its present and the test runs.

Reply:
I don't understand the answer for (1). The question is: "Why don't the other tests in this class need this same attribute?"

> Windows does have SQLite.Interop, so even though we aren't using that file its present and the test runs.

Is there a way we can ensure this, so we don't accidently lose test coverage?

Reply:
I could change the test so instead of skipping if it can't find the native dll, it skips if its not on x86/x64. Make an x86/x64 fact attribute.

Thoughts?

Reply:
I think that would be safer, if you want to make that change.

Reply:
Alright, that change has been made.